lib/widgets/lightbox.dart

```diff
@@ -258,9 +262,28 @@ class _ImageLightboxPageState extends State<_ImageLightboxPage> {
     // The full image is available, so display it.
     if (frame != null) return child;
 
+    // Use `BoxFit.contain` if the original image size is larger than
+    // the device screen size, otherwise `null` (default).
+    //
+    // This allows the thumbnail image to mimic the same behaviour as
+    // when the original image is loaded, which is, keep small images
+    // small (and in the center) and large images contained in the
+    // viewport.
```

Comment:
Let's add a test case or two for this behavior, since it seems like nontrivial logic.

I see we don't have any existing tests for the image lightbox. We don't have to add tests for the rest of its behavior now; but it'd be good to test this, and in particular to have a test exercising the situation I mentioned in another comment with a notched display.

If I find time today I can try to add a couple of basic tests for the existing layout of the image lightbox, to help set the framework for the new tests.

Reply:
Sent that as #835. So you can rebase this PR atop that one, and then add the new couple of test cases.